hydrolib/core/dflowfm/xyn/name_extrator.py

```diff
@@ -0,0 +1,77 @@
+class NameExtractor:
```

Comment:
Maybe this whole class could be replaced by the following regexp?
```python
prog=re.compile("^\s*('?)([^'#]+)\\1\s*#?.*$")
if (match := prog.match(text) is not None):
    return match.group(2)
# ...
```
See here some examples on which it works:
```
In [30]: print(prog.match("'somevalue in quotes' #comment").group(2))
somevalue in quotes

In [31]: print(prog.match("  somevalue_without_quotes #comment").group(2))
somevalue_without_quotes

In [32]: print(prog.match("  somevalue_without_quotes_and_no_comment  ").group(2))
somevalue_without_quotes_and_no_comment

In [33]: print(prog.match("'somevalue in quotes, but no comment'  ").group(2))
somevalue in quotes, but no comment

In [34]: print(prog.match("'somevalue in quotes, but no comment and missing closing quote  "))
None
```

Reply:
Upon second checking: no comments at the end of a data line are supported at all in .xyn files, so please remove the `#` handling (either in your own code, or in my above suggestion).

Reply:
I have spent some time trying to get it to work with a single regex, but I could not get it to work :(